bokeh/charts/_builder.py

```diff
@@ -258,64 +221,4 @@ def set_and_get(self, prefix, val, content):
         """
         self._set_and_get(self.data, prefix, self.attr, val, content)
 
-    @property
-    def palette(self):
```

Comment:
Ah, palette property was definitely something Ieft for later in the refactoring that I forgot (grrr, sorry). The idea was to    remove _set_colors in favor of palette.

Now, if I got it write, you've added cycle_colors which is used on all builders and palette is not used anyway. As _palette is set on the Builder.__init__ what about:

* changing `_palette` to `palette`
* change `self._palette = palette or DEFAULT_PALETTE` to `self.palette = palette or cycle_colors(...)`
* make all custom builders use self.palette instead of cycle_colors directly

?